FirebaseTestProject/ViewModels/AuthViewModel.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  AuthViewModel.swift
+//  AuthViewModel
+//
+//  Created by Tim Miller on 8/12/21.
+//
+
+
+import FirebaseAuth
+import SwiftUI
+
+@MainActor class AuthViewModel: ObservableObject {
+  
+  private let auth = Auth.auth()
+  
+  @AppStorage("userSignedIn") var userSignedIn: Bool = false
+  
+  func getUser() -> String {
+    guard let user = auth.currentUser?.email else {
+      return "No User"
+    }
+    return user
+  }
+  
+  func signIn(email: String, password: String) async {
+    do {
+      try await auth.signIn(withEmail: email, password: password)
+      UserDefaults.standard.set(true, forKey: "userSignedIn")
```

Comment:
LMK: Would this work if we removed the explicit setting of UserDefaults? I believe the `AppStorage` property wrapper should take care of this when the `userSignedIn` property is changed, but please correct me if I'm wrong!